internal/code/code_test.go

```diff
@@ -0,0 +1,79 @@
+package code_test
+
+import (
+	"testing"
+
+	"github.com/maaslalani/slides/internal/code"
+)
+
+func TestParse(t *testing.T) {
+	tt := []struct {
+		markdown string
+		expected code.Block
+	}{
+		// We can't put backticks ```
+		// in multi-line strings, ~~~ instead
+		{
+			markdown: `
+~~~ruby
+puts "Hello, world!"
+~~~
+`,
+			expected: code.Block{
+				Code:     `puts "Hello, world!"`,
+				Language: "ruby",
+			},
+		},
+		{
+			markdown: `
+~~~go
+fmt.Println("Hello, world!")
+~~~
+`,
+			expected: code.Block{
+				Code:     `fmt.Println("Hello, world!")`,
+				Language: "go",
+			},
+		},
+		{
+			markdown: `
+# Slide 1
+Just a regular slide, no code block
+`,
+			expected: code.Block{},
+		},
+		{
+			markdown: `
+# Multiple Code Blocks
+~~~go
+fmt.Println("Oh no!")
```

Comment:
I think for go code, I'm going to make this work with some first-class support.

We will it to a file (like @brittonhayes mentioned in #9) and add `package main` if it's not already there and wrap the code in `func main() { ... }` if it is not already and run `goimports` on the file.

That way this code will execute properly even though it is just a snippet of Go code. But that will be a separate issue.